pkg/cluster/sync.go

```diff
@@ -354,6 +357,26 @@ func (c *Cluster) syncStatefulSet() error {
 	return nil
 }
 
+// PropagateAnnotationsToStatefulsets updates annotations to statefulsets if required
+// based on the annotations in postgres CRD
+func (c *Cluster) PropagateAnnotationsToStatefulsets(annotations map[string]string) map[string]string {
+	ToPropagateAnnotations := c.OpConfig.StatefulsetPropagateAnnotations
+	PgCRDAnnotations := c.Postgresql.ObjectMeta.GetAnnotations()
+
+	if ToPropagateAnnotations != nil && PgCRDAnnotations != nil {
+		for _, anno := range ToPropagateAnnotations {
+			for k, v := range PgCRDAnnotations {
+				matched, err := regexp.MatchString(anno, k)
+				if err == nil && matched {
```

Comment:
Making it a bit more complex: Since regex can yield an error operator should validate regex on startup or log an error during comparison. Silent error here may hide problem too long.